src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -564,23 +557,17 @@ public void Dispose()
         {
             Debug.Assert(!handle.IsInvalid);
 
-            while (true)
+            long result = sysCall(handle, arg1, arg2, arg3);
```

Comment:
I've changed the checking to a `CheckPipeCall` method, and removed the delegate.